src/main/java/bio/terra/service/dataset/DatasetTableDao.java

```diff
@@ -176,45 +202,64 @@ private void removeColumns(UUID datasetId, UUID tableId, Collection<Column> colu
-
-          long bqPartitionVersion = rs.getLong("bigquery_partition_config_version");
-          String bqPartitionConfig = rs.getString("bigquery_partition_config");
-          if (bqPartitionVersion == 1) {
-            try {
-              table.bigQueryPartitionConfig(
-                  objectMapper.readValue(bqPartitionConfig, BigQueryPartitionConfigV1.class));
-            } catch (Exception ex) {
-              throw new CorruptMetadataException("Malformed BigQuery partition config", ex);
-            }
-          } else {
-            throw new CorruptMetadataException(
-                "Unknown BigQuery partition config version: " + bqPartitionVersion);
+              tableMap.computeIfAbsent(
+                  tableId,
+                  (id) -> {
+                    try {
+                      DatasetTable t =
+                          new DatasetTable()
+                              .id(id)
+                              .name(rs.getString("table_name"))
+                              .rawTableName(rs.getString("table_raw_table_name"))
+                              .softDeleteTableName(rs.getString("table_soft_delete_table_name"))
+                              .rowMetadataTableName(rs.getString("table_row_metadata_table_name"))
+                              .columns(new ArrayList<>())
+                              .primaryKey(new ArrayList<>());
+
+                      // Extract BigQuery partition config
+                      long bqPartitionVersion =
+                          rs.getLong("table_bigquery_partition_config_version");
+                      String bqPartitionConfig = rs.getString("table_bigquery_partition_config");
+                      if (bqPartitionVersion == 1) {
+                        try {
+                          t.bigQueryPartitionConfig(
+                              objectMapper.readValue(
+                                  bqPartitionConfig, BigQueryPartitionConfigV1.class));
+                        } catch (Exception ex) {
+                          throw new CorruptMetadataException(
+                              "Malformed BigQuery partition config", ex);
+                        }
+                      } else {
+                        throw new CorruptMetadataException(
+                            "Unknown BigQuery partition config version: " + bqPartitionVersion);
+                      }
+                      return t;
+                    } catch (SQLException e) {
+                      throw new RuntimeException(e);
+                    }
+                  });
+
+          // Add column to table
+          Column column = getColumnRowMapper(table).mapRow(rs, rowNum);
+          table.getColumns().add(column);
+
+          // Add primary key to table if appropriate
+          List<String> primaryKey = DaoUtils.getStringList(rs, "table_primary_key");
+          if (primaryKey.contains(column.getName())) {
+            table.getPrimaryKey().add(column);
           }
```

Comment:
It looks like this entire fragment is shared with `SnapshotTableDao`. I guess there's no good way to share code here?

Reply:
Not really in a way that doesn't make the code more confusing